### DISCUSSION TITLE
Strapi + Next.js in Nginx with SSL

### DISCUSSION THREAD

**Discussion by danielgymnasium** [strapi/strapi#6135] [Legacy Topics]
Hi,
i just got my server up and running and setup the SSL so far so good for my localhost:3000 everything is up and running and it serves the https however my img that come from localhost:1337 are been served as http, and i cannot go to myserver.com/admin 

i did add the location like so:

  `location / { `
      `    `      `    `           `    proxy_pass http://localhost:3000;`
  `    `      `    `                     `proxy_http_version 1.1;`
   `    `      `    `                `    proxy_set_header Upgrade $http_upgrade;`
    `    `      `    `                  `         proxy_set_header Connection 'upgrade';`
     `    `      `    `                        `  proxy_set_header Host $host;`
    `    `      `    `                   `        proxy_cache_bypass $http_upgrade;`
      `    }`
  `  location ^~ /api {`
  `    `      `    `     `      rewrite ^/api/(.*)$ /$1 break;`
  `    `      `    `       `    proxy_pass http://localhost:1337/;`
  `    }`
  `  location ^~ /admin {`
  `    `      `    `       `    rewrite ^/admin/(.*)$ /admin/$1 break;`
   `    `      `    `     `     proxy_pass http://localhost:1337/admin;`
  `    }`

i wonder if this is the right way, and what am i doing wrong ?

**Comment by derrickmehaffy** [comment]
A few things:

- Did you configure the proxy stuff in Strapi (server.json)
- Sub folder proxying isn't supported yet (PR already merged and this support will be in rc.1 version when released)

Alternative work around for now would be to use an upload provider like AWS-S3

**Reply by danielgymnasium** [reply]
No, I have not configure nothing proxy wise on strapi (server.json) only in nginx, what is it that i need to configure ? 

Will search mean while for it as well

**Reply by derrickmehaffy** [reply]
Yes you will need to let Strapi know about the proxy, we have added a guide to the docs relating to Strapi and Nginx: https://strapi.io/documentation/v3.x/deployment/nginx-proxy.html

**Comment by bamne123** [comment]
Hi I m using Docker for deployment, following Nginx Config I m using.
Hope this will help someone....

```
server {
    listen 80;
    server_name web.example.com;
    server_tokens off;

    location /.well-known/acme-challenge/ {
        root /var/www/certbot;
    }

    location / {
        return 301 https://$host$request_uri;
    }
}

server {
    listen 443 ssl;
    server_name web.example.com;
    server_tokens off;

    ssl_certificate /etc/letsencrypt/live/web.example.com/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/web.example.com/privkey.pem;
    include /etc/letsencrypt/options-ssl-nginx.conf;
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;

    location / {
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-NginX-Proxy true;
        proxy_pass http://frontend:3000;
        proxy_ssl_session_reuse off;
        proxy_set_header Host $http_host;
        proxy_cache_bypass $http_upgrade;
        proxy_redirect off;
    }
}

server {
    listen 80;
    server_name cms.example.com;
    server_tokens off;

    location /.well-known/acme-challenge/ {
        root /var/www/certbot;
    }

    location / {
        return 301 https://$host$request_uri;
    }
}

server {
    listen 443 ssl;
    server_name cms.example.com;
    server_tokens off;

    ssl_certificate /etc/letsencrypt/live/cms.example.com/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/cms.example.com/privkey.pem;
    include /etc/letsencrypt/options-ssl-nginx.conf;
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;

    location / {
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-NginX-Proxy true;
        proxy_pass http://backend:1337;
        proxy_ssl_session_reuse off;
        proxy_set_header Host $http_host;
        proxy_cache_bypass $http_upgrade;
        proxy_redirect off;
    }
}
```

**Reply by derrickmehaffy** [reply]
Thank you @bamne123 

For those still looking for help, please be sure to take a look at the docs for Nginx here: https://strapi.io/documentation/v3.x/deployment/nginx-proxy.html
